.ci/jenkins_job_build.sh

```diff
@@ -20,11 +20,13 @@ set -e
 cc_repo=$1
 
 tests_repo="github.com/clearcontainers/tests"
-mkdir -p ${HOME}/go
+
+# Use the Jenkins workspace for our work area
+export GOPATH=${WORKSPACE}/go
```

Comment:
Might be worth adding a check to ensure `WORKSPACE` is set? If not, either error, or fall back to the old directory with a message?

Reply:
Indeed, would not be a bad thing - I did briefly ponder what would happen if it were not set :-)
If we accept `WORKSPACE` is the right thing to do - and this file is titled 'jenkins', then I'd probably make it abort if not set.

Reply:
I agree with @jodh-intel  that we should fall back to `${HOME}/go` if the `WORKSPACE` does not exists as I (and maybe others) usually run the CI scripts on other machines to setup my CC environment.

Reply:
Sure, I'll add a check.
I'm not wholly sold on the fallback to `${HOME}`, mostly as the script is called `jenkins_job_build.sh` - thus inferring it is made for Jenkins. I know @sboeuf won't like renaming it and making it more 'jenkins neutral/unaware', as we'd have to modify all the jenkins setups to change the script name they are calling - but - something else that bugs me a little is that it is called `build`, yet it also blatantly does the run as well (I am forming a PR to stop it doing the run in the case of the metrics CI at least - but maybe we should lift the run out into another script, and wrap/rename this one...).

@sboeuf @chavafg thoughts on copy/splitting this script into two that do `build` and `run_qa`, and then making this a wrapper that invokes those two (or run_metrics for the metrics CI case).
I think that will then give us a clearer definition of what the scripts actually do and a touch more flexibility in how they are called either from different CI systems or by hand etc.

Reply:
and - I'm happy to put that PR together, of course

Reply:
@grahamwhaley well I gave this name to the script because it is executed in the `Build` section of the Jenkins job.

Reply:
BTW, we can do the fallback to `$HOME` but `$WORKSPACE` must be set in case of Jenkins. So I think you might consider to return an error if `$WORKSPACE` is not set.

Reply:
@grahamwhaley of course you can split the script, I don't mind (as long as we keep the name of this one as a wrapper as you mentioned). If you think splitting will make this clearer and more reusable, that's fine ;)

Reply:
How about erroring if `CI` is set and fallback if not?

Reply:
If the Jenkins script is going to change, could we maybe add a comment explaining why it has its name, and maybe throw in a reference to https://github.com/clearcontainers/jenkins.

Reply:
@jodh-intel Looks like this script sets `CI` :-) - and I don't see `CI` set in my Jenkins env. But, sure, we'll figure something out.

@chavafg just for clarity, to help me try to restructure and break up the scripts maybe - you run the scripts locally just to set up your CC env - you don't use then to actually run QA tests presumably?
That is sort of implying to me that we need some setup scripts for hand-running - and then I wonder how they are different from the distro install scripts we already have... your thoughts welcome here...

@mcastelino you were re-creating and running the CI locally recently I think - were you using Jenkins or running by hand (I suspect by hand) - and if so, did you use this script, and thus did you have to hack it or set up a bunch of env settings to make it work?

My plan is to see if we can break this apart to make it work for everybody (QA CI Jenkins, metrics CI Jenkins and hand-run cases) - and document that a bit.

Oh, and @jodh-intel yes, I intend to add more comments to the scripts to note why we have to do all the little bits in them we do (like why the test repo has its own little bit of magic etc.) - saves me having to work it all out again in another 6 months ;-)

Reply:
@grahamwhaley yes, it is to setup the CC environment only. Not sure what you mean with the distro install scripts? I use this script because it install latest versions of each component.

Reply:
@grahamwhaley I was not running the full jenkins setup. Just running what jenkins would run which was 
- Create VM
- Install Go
```
cd $GOHOME/src/github.com/clearcontainers/runtime/
sudo -E .ci/setup.sh
sudo -E .ci/run.sh
```
- comment out qemu-lite installation
- Switch branches
- Revendor virtcontainers into runtime (if virtcontainers was modified)